src/manifest_store.rs

```diff
@@ -146,7 +153,7 @@ impl StoredManifest {
     }
 
     pub(crate) async fn refresh(&mut self) -> Result<&CoreDbState, SlateDBError> {
-        let Some((id, manifest)) = self.manifest_store.read_latest_manifest().await? else {
+        let Some((id, manifest)) = self.manifest_store.try_read_latest_manifest().await? else {
             return Err(InvalidDBState);
```

Comment:
Guess it's better to keep this as `InvalidDbState` instead of `LatestManifestMissing` since we expect a latest manifest here...